File: java-programming/examples/shapes/Point.java
public class Point {
    private int x;
    private int y;

    public Point(int a, int b) {
        x = a;
        y = b;
    }

    public String showCoord() {
        return "(" + x + ", " + y + ")";
    }

    public void move(int dx, int dy) {
        x += dx;
        y += dy;
    }

    public int getx() {
        return x;
    }
    public int gety() {
        return y;
    }
}

File: books/perl-oop/src/examples/perloop/point/eg/point.pl
use strict;
use warnings;

use FindBin;
use lib "$FindBin::Bin/../lib";


use Point;

my $p = Point->new();

$p->set_x( 23 );
$p->set_y( 12 );
print $p->get_x, "\n";
print $p->get_y, "\n";

my $q = Point->new(x => 10, y => 20);

print $q->get_x, "\n";
print $q->get_y, "\n";
